src/chainstate/stacks/db/transactions.rs

```diff
@@ -2148,6 +2173,90 @@ pub mod test {
         }
     }
 
+    #[test]
+    fn process_smart_contract_transaction_syntax_error() {
+        let contracts = [
+            "(define-data-var bar int 0)) ;; oops",
+            ";; `Int` instead of `int`
+             (define-data-var bar Int 0)",
```

Comment:
suggestion to include tests for both analysis & parse errors

Reply:
The first test here is a parse error and the second test is an analysis error.